src/shared/seccomp-util.c

```diff
@@ -101,43 +147,50 @@ int seccomp_init_conservative(scmp_filter_ctx *ret, uint32_t default_action) {
         return r;
 }
 
-int seccomp_add_secondary_archs(scmp_filter_ctx c) {
+#define ADD_ARCH(arch)                          \
+        r = seccomp_arch_add(c, arch);          \
+        if (r < 0 && r != -EEXIST)              \
+                return r
```

Comment:
i don't like macros like this very much. I think defining a static const array of the arch ids that matter here (even if it only has one or two entries on typical archs) would be much better, and then iterating through that. and more efficient in some cases probably too...